### PR TITLE
qemu: Treat qxl as an unsupported backend

### PR DESCRIPTION
**Pull request by kalikiana**
- Setting QEMUVGA to qxl is considered unsupported.
- Add unit test coverage for -vga.

See: https://progress.opensuse.org/issues/111992

### CODE REVIEW

**Comment by Martchus** [comment]
So there was no way to make qxl use the right resolution?

File: backend/qemu.pm
Comment (okurz) [inline backend/qemu.pm:683]: Please don't set that. Some time ago I have explicitly removed the default to leave it to qemu to select a default. That fixed some problems we had

**Review by okurz — CHANGES_REQUESTED** [review]

**Comment by kalikiana** [comment]
> So there was no way to make qxl use the right resolution?

The only way I'm aware of is by interactively going through the menu.

**Comment by okurz** [comment]
> > So there was no way to make qxl use the right resolution?
> 
> The only way I'm aware of is by interactively going through the menu.

That's what we discussed, yes. I suggest to do this then.

**Comment by kalikiana** [comment]
> > The only way I'm aware of is by interactively going through the menu.
> 
> That's what we discussed, yes. I suggest to do this then.

I wasn't planning on looking into that because it requires steps to be executed in the test rather than backend configuration. The PR fulfills the AC because `qxl is not used anymore`.

**Comment by okurz** [comment]
> The PR fulfills the AC because `qxl is not used anymore`.

Yes, it would. But I will only merge this if test maintainers can confirm that they can live without qxl.

@Vogtinator do you think we can switch to alternative QEMUVGA selections in tests or do you consider selecting a resolution within the OVMF UEFI menu – just like the boot device – a better choice?

**Comment by Vogtinator** [comment]
IMO just not supporting `qxl` is not a valid solution. It is currently used in some tests and we'd lose some coverage there and at the very least would have to modify the windows images to be able to switch those tests to virtio instead.

It's also not just `qxl` that's affected by the default resolution change. It's all video devices, except for a few exceptions like `cirrus`.

What we could try to do is perform some OVMF configuration manually and then use the resulting vars flash file in tests. This would probably have to be repeated for every OVMF update though.

**Comment by Martchus** [comment]
Let me try convincing `qxl` to use our default resolution via some qemu params.

**Comment by Martchus** [comment]
If I just clone https://openqa.opensuse.org/tests/2400280 with `QEMUVGA=qxl` everything works as expected. So that's not good enough for reproducing the issue as it is likely additionally also OVMF-specific. If I clone https://openqa.opensuse.org/tests/2400506 (the job from the issue) then it fails locally with `runcmd '/usr/bin/qemu-img create -f qcow2 -F qcow2 -b /hdd/openqa-devel/openqa/pool/1/windows-10-x86_64-21H1@uefi_win.qcow2 /hdd/openqa-devel/openqa/pool/1/raid/hd0-overlay0 110080' failed with exit code 1: 'qemu-img: /hdd/openqa-devel/openqa/pool/1/raid/hd0-overlay0: Image is not in qcow2 format`. I still need to investigate why that's the case.

**Comment by Vogtinator** [comment]
Yes, the OVMF resolution should mostly impact OVMF (shell and menu) as well as bootloaders. The kernel will also inherit this resolution until the "real" driver is loaded and new configuration applied.

**Review by Martchus — DISMISSED** [review]
Looks like `-device qxl,xres=1024,yres=768` works just fine. Let me create a PR. If it is acceptable I'd close this one.

**Comment by Vogtinator** [comment]
> Looks like `-device qxl,xres=1024,yres=768` works just fine. Let me create a PR. If it is acceptable I'd close this one.

Did you also use `-vga none`? If not, you unintentionally got both `-vga std` and the qxl device attached, with vga std being the default device.

File: backend/qemu.pm
Comment (Martchus) [inline backend/qemu.pm:685]: A fatal `die` would overshoot the goal. "Unsupported" doesn't mean "not allowed at all". As shown in https://github.com/os-autoinst/os-autoinst/pull/2082#issuecomment-1152199464 it is also not *that* bad, e.g. if one would get past the bootloader it would actually work.

File: backend/qemu.pm
Comment (Martchus) [inline backend/qemu.pm:686]: `QEMUVGA` might be `undef` here.

**Review by Martchus — CHANGES_REQUESTED** [review]

File: backend/qemu.pm
Comment (kalikiana) [reply backend/qemu.pm:685]: Well, we don't know what's installed and wether the test has needles for the right resolution.

**Pull request closed without merging**
